### PR TITLE
fix(backend): various perf optimizations

### PR DESCRIPTION
**Pull request by fatih-acar**
This should improve read paths when using the following query against the demo data
We should actually run this kind of query during the CI benchmark step but that requires loading the demo data...

```
query {
  InfraDevice {
    edges {
      node {
        name {
          value
        }
        interfaces {
          edges {
            node {
              name {
                value
              }
            }
          }
        }
      }
    }
  }
}
```

### CODE REVIEW

File: python_sdk/infrahub_sdk/uuidt.py [hunk 1]
```diff
@@ -39,9 +41,22 @@ def __init__(
         hostname: Optional[str] = None,
         random_chars: Optional[str] = None,
     ):
-        self.namespace = namespace or str(Path(__file__).parent.resolve())
```
Comment (dgarros) [inline python_sdk/infrahub_sdk/uuidt.py:42]: Are you seeing some performance issue related to the uuidt ?

Comment (fatih-acar) [reply python_sdk/infrahub_sdk/uuidt.py:42]: Yes, I'm trying to get rid of the following regression: https://github.com/opsmill/infrahub/pull/3541

Comment (ogenstad) [reply python_sdk/infrahub_sdk/uuidt.py:42]: Seems the problem is that `str(Path(__file__).parent.resolve()` is evaluated each time we create a class.

Shouldn't it be a global variable as:

```python
DEFAULT_NAMESPACE=str(Path(__file__).parent.resolve())
```
Then within the function we leave it as it was with:

```python
self.namespace = namespace or GLOBAL_NAMESPACE
```

That way we only do the Path lookup once.

Comment (ogenstad) [reply python_sdk/infrahub_sdk/uuidt.py:42]: Same with the gethostname()

Comment (fatih-acar) [reply python_sdk/infrahub_sdk/uuidt.py:42]: Indeed... thanks for the tip, I become a python newbie sometimes :)

**Review by dgarros — APPROVED** [review]

**Pull request merged**